include/DeterministicNoiseSimulator.hpp

```diff
@@ -3,124 +3,132 @@
             throw std::runtime_error("Error probabilities are faulty!"
                                      "\n single qubit error probability: " +
                                      std::to_string(noiseProbSingleQubit) +
                                      " multi qubit error probability: " + std::to_string(noiseProbMultiQubit) +
                                      "\n single qubit amplitude damping  probability: " + std::to_string(ampDampingProbSingleQubit) +
                                      " multi qubit amplitude damping  probability: " + std::to_string(ampDampingProbMultiQubit));
         }
+        Simulator<DDPackage>::dd->resize(qc->getNqubits());
     }
 
     std::map<std::string, std::size_t> Simulate([[maybe_unused]] unsigned int shots) override {
-        return {};
-    };
-
-    std::map<std::string, double> DeterministicSimulate();
+        auto m = this->DeterministicSimulate();
 
-    [[nodiscard]] std::string intToString(long target_number, char value) const;
+        auto k = sampleFromProbabilityMap(m, shots);
 
-    void applyDetNoiseSequential(const qc::Targets& targets);
+        return sampleFromProbabilityMap(m, shots);
+    };
 
-    [[nodiscard]] std::map<std::string, double> AnalyseState(dd::QubitCount nr_qubits, bool full_state);
+    std::map<std::string, dd::fp> DeterministicSimulate();
 
-    void setNoiseEffects(const std::string& cGateNoise) {
-        for (const auto& effect: cGateNoise) {
-            if (effect != 'A' && effect != 'P' && effect != 'D') {
-                throw std::runtime_error("Unknown noise operation '" + std::to_string(effect) + "'\n");
-            }
-        }
-        gateNoiseTypes = cGateNoise;
-    }
+    std::map<std::string, std::size_t> sampleFromProbabilityMap(const std::map<std::string, dd::fp>& resultProbabilityMap, unsigned int shots);
 
     [[nodiscard]] dd::QubitCount getNumberOfQubits() const override { return qc->getNqubits(); };
 
     [[nodiscard]] std::size_t getNumberOfOps() const override { return qc->getNops(); };
 
     [[nodiscard]] std::string getName() const override { return qc->getName(); };
 
-    const std::map<char, int> gateNoiseEffects = {
-            {'B', 2}, //Bit-flip
-            {'P', 2}, //Phase-flip
-            {'A', 2}, //Amplitude Damping
-            {'D', 4}, //Depolarisation
-    };
-
-    dEdge density_root_edge{};
+    [[nodiscard]] std::size_t getActiveNodeCount() const override { return Simulator<DDPackage>::dd->dUniqueTable.getActiveNodeCount(); }
+    [[nodiscard]] std::size_t getMaxNodeCount() const override { return Simulator<DDPackage>::dd->dUniqueTable.getMaxActiveNodes(); }
+    [[nodiscard]] std::size_t getMaxMatrixNodeCount() const override { return Simulator<DDPackage>::dd->mUniqueTable.getMaxActiveNodes(); }
+    [[nodiscard]] std::size_t getMatrixActiveNodeCount() const override { return Simulator<DDPackage>::dd->mUniqueTable.getActiveNodeCount(); }
```

Comment:
why are these overridden here if they just perform the same action as in the base class?

Reply:
Removed redundant lines